compiler/src/iree/compiler/Dialect/Modules/HAL/Loader/IR/test/BUILD

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+load("//build_tools/bazel:iree_lit_test.bzl", "iree_lit_test_suite")
+load("//build_tools/bazel:enforce_glob.bzl", "enforce_glob")
+
+package(
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+iree_lit_test_suite(
+    name = "lit",
+    srcs = enforce_glob(
+        [
+        ],
+        include = ["*.mlir"],
```

Comment:
Tests TODO? I was specifically wondering how

```c++
void ExecutableLoadOp::getAsmResultNames(
    function_ref<void(Value, StringRef)> setNameFn) {
  setNameFn(getResult(), "exe");
}

//===----------------------------------------------------------------------===//
// hal_loader.executable.lookup
//===----------------------------------------------------------------------===//

void ExecutableLookupOp::getAsmResultNames(
    function_ref<void(Value, StringRef)> setNameFn) {
  setNameFn(getResult(), "exe");
}
```

works (should those `"exe"`s be different between the two ops?)

Reply:
That's standard MLIR stuff and just for setting the base name of SSA values - cosmetic only. Multiple ops can name things the same as they all get a suffix (`%exe_0` `%exe_1` `%exe_2` etc)

Reply:
Added a test for the only non-boilerplate code in IR (a dispatch folder)